packages/components/src/templates/next/render/renderComponentPreviewText.ts

```diff
@@ -0,0 +1,69 @@
+import Ajv from "ajv"
+
+import type { ProseContent, TextProps } from "~/interfaces"
+import type { IsomerSchema } from "~/types"
+import { TextSchema } from "~/interfaces"
+
+function getTextContentOfProse(content: ProseContent): string {
+  const values: string[] = []
+
+  function recursiveSearch(obj: Record<string, unknown>) {
```

Comment:
i haven't tried it yet, but `editor.getText` (see [here](https://tiptap.dev/docs/editor/api/editor#gettext)) seems ideal for this - should we do it on studio since this appears to be studio specific?

Reply:
we aren't using the tiptap editor for this part yet

Reply:
nono, i mean since this part comes from tiptap, we can shift this over to tiptap

Reply:
oh as in, current implementation is to put it in Component library which don't have tiptap installed

Reply:
eh bad phrasing on my end, i meant since this part comes from tiptap, we can shift over to studio!